### PR TITLE
Update Docker Registry workflow examples for docker/build-push-action@v2

### LINKED ISSUE #808: Update docker/build-push-action in "Publishing Docker images"
### What article on docs.github.com is affected?

[Publishing Docker images](https://docs.github.com/en/free-pro-team@latest/actions/guides/publishing-docker-images#publishing-images-to-github-packages) (and [markdown file](https://github.com/github/docs/blob/main/content/actions/guides/publishing-docker-images.md) on GitHub)

### What part(s) of the article would you like to see updated?

Update usage of `docker/build-push-action` to v2. The docs for build-push-action are very different from the usage listed here so it's not clear how v2 should be used, if at all.

### PR DESCRIPTION
**Pull request by rajivshah3**
### Why:
Docker released a new major version of their [`build-push-action`](https://github.com/docker/build-push-action), but it's not immediately clear how to migrate some of the functionality used in the examples in the GitHub Docs. This should make it a bit easier to use the v2 version.

The current docs look like this:
- https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#publishing-a-package-using-an-action
- https://docs.github.com/en/actions/guides/publishing-docker-images#publishing-images-to-github-packages

Closes #808 (already closed)

### What's being changed:

The example workflows and descriptions for publishing Docker images now uses the new version of the `build-push-action` combined with Docker's `login-action`

### Check off the following:

- [x] I have reviewed my changes in staging (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (eg. a "before and after video")

### CODE REVIEW

**Comment by janiceilene** [comment]
@rajivshah3 Thanks so much for opening a PR and giving so much context about why you feel this is an important change! I'm going to triage this for review, but can you resolve the merge conflicts in the meantime?

File: content/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions.md [hunk 1]
```diff
@@ -248,114 +252,84 @@ on:
...
 {% raw %}
   ```yaml
-username: ${{ github.actor }}
+  - name: Build container image
   ```
 {% endraw %}
   </td>
   <td>
-    Defines the user account that will publish the packages. Once published, the packages are owned by the account defined here.
+    Creates a new step called <code>Build container image</code>. This step runs as part of the <code>build-and-push-image</code> job. The <code>needs: run-npm-test</code> command makes this job dependent on the <code>run-npm-test</code> job.
```
Comment (skedwards88) [inline content/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions.md:278]: These changes are great! I'm going to push a change that modifies this table a bit to talk about the `needs` statement higher up.

**Review by skedwards88 — COMMENTED** [review]
✨ Thank you for these detailed updates!!! I just had a few small wording suggestions and an addition that I will incorporate.

**Review by skedwards88 — APPROVED** [review]
Thank you again for this work! I'll get this merged down for you.

**Review by demianstack6 — COMMENTED** [review]
Damiangallegos40@gmail.com
gallegosdamian-
damiangallegos+

**Pull request merged**